pkg/agent/config.go

```diff
@@ -89,6 +92,33 @@ type VenafiCloudConfig struct {
 	UploadPath string `yaml:"upload_path,omitempty"`
 }
 
+// MachineHubConfig holds configuration values specific to the CyberArk Machine Hub integration
+type MachineHubConfig struct {
+	// Subdomain is the subdomain indicating where data should be pushed. Used
+	// for querying the Service Discovery Service to discover the Identity API
+	// URL.
+	Subdomain string `yaml:"subdomain"`
+
+	// CredentialsSecretName is the name of a Kubernetes Secret in the same
+	// namespace as the agent, which will be watched for a username and password
+	// to send to CyberArk Identity for authentication.
+	CredentialsSecretName string `yaml:"credentialsSecretName"`
```

Comment:
suggestion: do you think we could include the secret watcher in this PR, or should that be left for a separate PR? It's definitely something we'll need!

Reply:
I'll add the watcher to the next PR 😅